pints/_mcmc/_adaptive.py

```diff
@@ -26,27 +28,190 @@ class AdaptiveCovarianceMCMC(pints.MCMC):
+        Set MCMC routine required variables as default.
+
+        # Target acceptance rate
+        acceptance_target = 0.25
+        # Total number of iterations
+        # Default set for small iteration (linear grow)
+        iterations = 2000 * dimension
+        # Number of iterations before using adapation
+        adaptation = 0.25 * iterations
+        # Number of iterations to discard as burn-in
+        # Only make sense for small iterations
+        burn_in = 0.5 * iterations
+        # Thinning: Store only one sample per X
+        thinning = 4
+        """
+        # Target acceptance rate
+        self._acceptance_target = 0.25
+        # Total number of iterations
+        self._iterations = self._dimension * 2000
+        # Number of iterations before adapation
+        self._adaptation_rate = 0.25
+        self._adaptation = int(self._iterations * self._adaptation_rate)
+        # Number of iterations to discard as burn-in
+        self._burn_in_rate = 0.5
+        self._burn_in = int(self._iterations * self._burn_in_rate)
+        # Thinning: Store only one sample per X
+        self._thinning = 4
+    
+    def acceptance_rate(self):
+        """
+        Return target acceptance rate.
+        """
+        return self._acceptance_target
+
+    def iterations(self):
+        """
+        Return the total number of iterations.
+        """
+        return self._iterations
+
+    def adaptation(self):
+        """
+        Return the number of iterations before adapation.
+        """
+        return self._adaptation
+
+    def adaptation_rate(self):
+        """
+        Return the first fraction of the total number of iterations before
+        using adaptation.
+        """
+        return self._adaptation_rate
+
+    def burn_in(self):
+        """
+        Return the number of iterations to discard as burn-in.
+        """
+        return self._burn_in
+
+    def burn_in_rate(self):
+        """
+        Return the first fraction of the total number of iterations to discard as
+        burn-in.
+        """
+        return self._burn_in_rate
+
+    def thinning(self):
+        """
+        Return thinning.
+        """
+        return self._thinning
+
+    def set_acceptance_rate(self, rate):
+        """
+        Set target acceptance rate.
+        """
+        self._acceptance_target = rate
+
+    def set_iterations(self, iterations):
+        """
+        Set the total number of iterations.
+        """
+        self._iterations = iterations
+        self._adaptation = int(self._iterations * self._adaptation_rate)
+        self._burn_in = int(self._iterations * self._burn_in_rate)
+        if self._thinning > self._adaptation:
+            print('WARNING: Thinning larger than adaptation')
+        if self._thinning > self._burn_in:
+            print('WARNING: Thinning larger than burn-in')
+        if self._thinning > self._iterations:
+            raise Exception('Thinning larger than total number of iterations')
```

Comment:
A bare `Exception` is almost never appropriate. You can look up good error types here: https://docs.python.org/2/library/exceptions.html. In this case, a `ValueError` would be appropriate (but see above!)